iOS/DuckDuckGoTests/SwitchBar/SwitchBarHandlerTests.swift

```diff
@@ -0,0 +1,391 @@
+//
+//  SwitchBarHandlerTests.swift
+//  DuckDuckGoTests
+//
+//  Copyright © 2025 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import XCTest
+@testable import Core
+@testable import DuckDuckGo
+import Combine
+import PersistenceTestingUtils
+
+final class SwitchBarHandlerTests: XCTestCase {
+
+    private enum StorageKey {
+        static let toggleState = "SwitchBarHandler.toggleState"
+    }
+
+    private var sut: SwitchBarHandler!
```

Comment:
For transparency. I've generated some test cases manually and asked duck.ai to generate the other permutations :)